Source/Core/VideoBackends/OGL/TextureCache.cpp

```diff
@@ -589,26 +539,31 @@ void TextureCache::CopyEFBToCacheEntry(TCacheEntry* entry, bool is_depth_copy,
 
   glViewport(0, 0, destination_texture->GetConfig().width, destination_texture->GetConfig().height);
 
-  GLuint uniform_location;
-  if (is_depth_copy)
-  {
-    m_depthMatrixProgram.Bind();
-    if (s_DepthCbufid != cbuf_id)
-      glUniform4fv(m_depthMatrixUniform, 5, colmat);
-    s_DepthCbufid = cbuf_id;
-    uniform_location = m_depthCopyPositionUniform;
-  }
-  else
+  auto uid = GetTextureConverterShaderUid(dst_format, is_depth_copy, is_intensity, scale_by_half);
+
+  auto it = m_efb_copy_programs.emplace(uid, EFBCopyShader());
+  EFBCopyShader& shader = it.first->second;
+  bool created = it.second;
+
+  if (created)
   {
-    m_colorMatrixProgram.Bind();
-    if (s_ColorCbufid != cbuf_id)
-      glUniform4fv(m_colorMatrixUniform, 7, colmat);
-    s_ColorCbufid = cbuf_id;
-    uniform_location = m_colorMatrixPositionUniform;
+    ShaderCode code = GenerateTextureConverterShaderCode(APIType::OpenGL, uid.GetUidData());
+
+    const std::string geo_program = g_ActiveConfig.iStereoMode > 0 ? geometry_program : "";
+
+    const char* prefix = geo_program.empty() ? "f" : "v";
```

Comment:
You could change the formatting specifier from a null-terminated string into just formatting a single char with `%c`, no?